modules/library/main/src/main/java/org/geotools/filter/function/FilterFunction_geometry.java

```diff
@@ -0,0 +1,74 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2015, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.filter.function;
+
+import org.geotools.filter.FunctionExpressionImpl;
+import org.geotools.filter.capability.FunctionNameImpl;
+import org.opengis.feature.Feature;
+import org.opengis.feature.GeometryAttribute;
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.filter.capability.FunctionName;
+import org.opengis.filter.expression.VolatileFunction;
+
+import com.vividsolutions.jts.geom.Geometry;
+
+/**
+ * Function the returns the default geometry of a feature, or null if there is none, or it's not a
+ * JTS geometry
+ * 
+ * @author Andrea Aime - GeoSolutions
+ */
+public class FilterFunction_geometry  extends FunctionExpressionImpl implements VolatileFunction {
```

Comment:
This naming convention is based on a code generator, for functions we make by hand GeometryFunction would be a better choice.